File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution 
{
    public List<List<Integer>> generate(int numRows) 
    {
        List<List<Integer>> A = new ArrayList<>();
        for (int i = 0; i < numRows; i++) 
        {
            List<Integer> B = new ArrayList<>();
            for (int j = 0; j < i + 1; j++) 
            {
                if (j == 0 || j == i) 
                {
                    B.add(1);
                } 
                else 
                {
                    B.add(A.get(i - 1).get(j - 1) + A.get(i - 1).get(j));
                }
            }
            A.add(B);
        }
        return A;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution(object):
    def generate(self, numRows):
        A = []
        for i in range (numRows):
            B = []
            for j in range (i + 1):
                if (j == 0 or j == i):
                    B.append(1)
                else:
                    B.append(A[i - 1][j - 1] + A[i - 1][j])
            A.append(B)    
        return A    